src/ServiceControl.AcceptanceTests/Recoverability/MessageFailures/FailedMessageRetriesController.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace ServiceControl.AcceptanceTests.Recoverability.MessageFailures
+{
+    using System;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Web.Http;
+    using Infrastructure.WebApi;
+    using Operations;
+    using Raven.Client;
+    using ServiceControl.Recoverability;
+
+    public class FailedMessageRetriesCountReponse
+    {
+        public int Count { get; set; }
+    }
+
+    public class FailedMessageRetriesController : ApiController
+    {
+        internal FailedMessageRetriesController(IDocumentStore store)
+        {
+            this.store = store;
+        }
+
+        [Route("failedmessageretries/count")]
+        [HttpGet]
+        public async Task<HttpResponseMessage> GetFailedMessageRetriesCount()
+        {
+            using (var session = store.OpenAsyncSession())
+            {
+                var query =
+                    session.Query<FailedMessageRetry>().Statistics(out var stats);
```

Comment:
Will this reuse the `FailedMessageRetries/ByBatch` index, or create a new one? Or is it assumed that the volume will be low enough that it doesn't matter?

Though I don't totally get why a controller is in the acceptance tests in the first place. Do we not care because it's just for a test?

Also, the `stats` should have `TotalResults` so it shouldn't be necessary to enumerate the results with `CountAsync()`. Though I see this is also in use in FailedErrorsController.

Reply:
> Will this reuse the FailedMessageRetries/ByBatch index, or create a new one? Or is it assumed that the volume will be low enough that it doesn't matter?

Probably create a new one? It's only for the tests and only the actual test instances that use this controller, so it should be fine.

> Though I don't totally get why a controller is in the acceptance tests in the first place. Do we not care because it's just for a test?

I'm following the convention set by the `FailedErrorsController`, I assume the thinking behind it is so that the tests don't have to have raven code directly in them because reasons?

> Also, the stats should have TotalResults so it shouldn't be necessary to enumerate the results with CountAsync(). Though I see this is also in use in FailedErrorsController.

Updated.

Reply:
Part of it is a bit legacy reasons. Since the beginning it WA possible to bring in Nancy modules and when we switched to Web api controllers that run inside the instance and therefore have access to everything available on the container. The tests have always used a wired up http client to communicate via the http api with SC and I assume that's why the obvious way expose internal details was also using http by injecting controllers. That being said though I'm pretty sure the new acceptance test base magic I wrote would allow to expose the doc store client directly but probably overkill and not worth diverging from the current way plus I actually like that all the ATTs use one consistent way to interact with the instance